### PR TITLE
fix: use 4 spaces for indentation in macro expansion

### LINKED ISSUE #16471: inline assist not working properly on `Hash::hash`
**rust-analyzer version**: 0.4.1810-standalone

**rustc version**: rustc 1.75.0 (82e1608df 2023-12-21)

**relevant settings**: None

with this code,
```rust
pub struct MyStruct {
    value: u64,
}

impl std::hash::Hash for MyStruct {
    fn hash<H: std::hash::Hasher>(&self, state: &mut H) {
        self.value.hash(state);
    }
}

impl MyStruct {
    pub fn as_u64(&self) -> u64 {
        self.value
    }
}
```

place the cursor on `hash` in `self.value.hash(state);` and apply the inline function assist

this is the result:
```rust
...
        {
            let this = &self.value;
            let state: &mutH = state;
          state.write_u64(*self)
        };
...
```

- `&mutH` should be `&mut H`
- `*self` should be `*this`
- the first two lines are indented with 4 spaces (normal), but `state.write_u64(*self)` only with 2, this is automatically fixed with the formatter but still doesnt look nice

the cause could be that the `Hash` impl for `u64` is macro generated, but i dont know exactly

### PR DESCRIPTION
**Pull request by roife**
Partial fix for #16471.
 
In the previous code, the indentation produced by macro expansion was set to 2 spaces. This PR modifies it to 4 spaces for the sake of consistency.

### CODE REVIEW

File: crates/ide-assists/src/handlers/inline_call.rs [hunk 1]
```diff
@@ -358,16 +358,15 @@ fn inline(
...
-                })
-                .for_each(|usage| {
-                    ted::replace(usage, &this());
-                });
-        }
+
+        body.syntax()
+            .descendants_with_tokens()
+            .filter_map(NodeOrToken::into_token)
+            .filter(|tok| tok.kind() == SyntaxKind::SELF_KW)
```
Comment (Veykril) [inline crates/ide-assists/src/handlers/inline_call.rs:365]: That approach won't work, as `self` can also refer to the current module

**Comment by Veykril** [comment]
I don't think we can do much about `1.` right now as our search infra is just not built in a way that allows to search like that

File: crates/ide-assists/src/handlers/inline_call.rs (hunk 1 above)
Comment (roife) [reply crates/ide-assists/src/handlers/inline_call.rs:365]: You are right, I forgot the other semantics of `self` 😱

**Comment by roife** [comment]
I agree that replacement of `self` keyword is non-trivial since it may refer the 'current module'. I have changed the title and the description of this PR.

**Review by Veykril — APPROVED** [review]

**Comment by Veykril** [comment]
Thanks!
@bors r+

**Pull request merged**